io.github.nearinfinitybrowser.nearinfinity.yml

```diff
@@ -0,0 +1,64 @@
+app-id: io.github.nearinfinitybrowser.nearinfinity
+runtime: org.freedesktop.Platform
+runtime-version: '22.08'
+sdk: org.freedesktop.Sdk
+sdk-extensions:
+  - org.freedesktop.Sdk.Extension.openjdk17
+command: nearinfinity
+
+finish-args:
+  - --share=ipc
+  - --socket=x11
+  - --socket=pulseaudio
+  - --socket=cups
+  # Required to access game folders outside of $HOME
+  - --filesystem=host
+  # Required to make game assets temporarily available to associated applications (e.g. video files)
+  - --filesystem=/tmp
+  - --env=PATH=/usr/bin:/app/bin:/app/jre/bin
+  - --env=JAVA_HOME=/app/jre
+
+modules:
+  - name: openjdk
+    buildsystem: simple
+    build-commands:
+      - /usr/lib/sdk/openjdk17/install.sh
+
+  - name: nearinfinity
+    buildsystem: simple
+    build-commands:
+      - install -Dm755 nearinfinity "${FLATPAK_DEST}/bin/nearinfinity"
+      - install -Dm644 NearInfinity.jar "${FLATPAK_DEST}/share/NearInfinity.jar"
+      - install -Dm644 "${FLATPAK_ID}.metainfo.xml" "${FLATPAK_DEST}/share/metainfo/${FLATPAK_ID}.metainfo.xml"
+      - install -Dm644 "${FLATPAK_ID}.desktop" "${FLATPAK_DEST}/share/applications/${FLATPAK_ID}.desktop"
+      - install -Dm644 App256.png "${FLATPAK_DEST}/share/icons/hicolor/256x256/apps/${FLATPAK_ID}.png"
+      - install -Dm644 io.github.nearinfinitybrowser.nearinfinity.xml "${FLATPAK_DEST}/share/mime/packages/io.github.nearinfinitybrowser.nearinfinity.xml"
+      - |
+        for dim in 16 22 32 64 128 256; do
```

Comment:
since the icons don't come from upstream you should keep the minimum, ie only 256x256